src/libs/client.js

```diff
@@ -123,14 +128,28 @@ export const fetchCozyDataForSlug = async (slug, client, cookie) => {
       }
     : undefined
 
-  const result = await stackClient.fetchJSON(
-    'GET',
-    `/apps/${slug}/open`,
-    undefined,
-    options
-  )
+  try {
+    const result = await stackClient.fetchJSON(
+      'GET',
+      `/apps/${slug}/open`,
+      undefined,
+      options
+    )
+
+    storeClientCachedData(client, cacheKey, result)
 
-  return result
+    return result
+  } catch (err) {
+    if (err.message === 'Network request failed') {
+      const cachedResult = await getClientCachedData(client, cacheKey)
+
+      if (cachedResult) {
+        return cachedResult
+      }
+    }
+
+    throw err
+  }
```

Comment:
So this cache has an unlimited lifetime ?

Reply:
For now, yes it is. I imagine that opening the app after a very long time AND while being offline is a rare event so this should be acceptable for now. But for sure we may want to add an invalidation mechanism soon.